lighthouse-core/gather/gatherers/js-usage.js

```diff
@@ -34,9 +43,23 @@ class JsUsage extends Gatherer {
     /** @type {Record<string, Array<LH.Crdp.Profiler.ScriptCoverage>>} */
     const usageByUrl = {};
     for (const scriptUsage of scriptUsages) {
-      const scripts = usageByUrl[scriptUsage.url] || [];
+      // ScriptCoverages without a URL are from code we run over the protocol.
```

Comment:
I _think_ it's a little bigger than that. I presume url-less scripts are generally "anonymous scripts"..  
[Looking at the chromium source](https://source.chromium.org/chromium/chromium/src/+/master:v8/src/inspector/v8-profiler-agent-impl.cc;l=392-401;drc=fa3aada528577aab1ad9a282108e44bf318310bf).. yeah this appears to be any script that doesnt have a sourceURL or a resolvable URL.  and yeah that's anonymous. 

anon scripts include `eval()`, `new Function`, `onload="..."` etc. I guess the `Runtime.evaluate` payloads end up as anonymous too.

Anyway, anonymous scripts dont give us much attribution to report to the user. 
so i'm fine filtering them out.

can you tweak the comment to capture this? ^

Reply:
makes sense, and matches the ScriptParsedEvents I see from store.google.com.

I think we should go a bit further and also `continue` if `scriptParsedEvent.embedderName === ''`, at least for now. We don't have a way to use coverage data for anonymous functions in the audit anyway, so collecting is pointless atm, although i suppose it doesn't hurt .... wdyt?

Reply:
> I think we should go a bit further and also `continue` if `scriptParsedEvent.embedderName === ''`

yup sg